lib/services/compare.go

```diff
@@ -24,16 +24,18 @@ import (
 	"github.com/google/go-cmp/cmp"
 	"github.com/google/go-cmp/cmp/cmpopts"
 
+	headerv1 "github.com/gravitational/teleport/api/gen/proto/go/teleport/header/v1"
 	"github.com/gravitational/teleport/api/types"
 )
 
 // CompareResources compares two resources by all significant fields.
-func CompareResources(resA, resB types.Resource) int {
+func CompareResources[T any](resA, resB T) int {
 	equal := cmp.Equal(resA, resB,
 		ignoreProtoXXXFields(),
```

Comment:
Instead of this could we use protocmp.Transform? https://pkg.go.dev/google.golang.org/protobuf/testing/protocmp#Transform

Reply:
Fair enough, I appreciate the attempt.